core/src/mindustry/type/UnitType.java

```diff
@@ -145,6 +145,14 @@ public Unit spawn(float x, float y){
         return spawn(state.rules.defaultTeam, x, y);
     }
 
+    public Unit spawn(Team team, Posc pos){
+        return spawn(team, pos.x, pos.y);
+    }
+
+    public Unit spawn(Posc pos){
+        return spawn(state.rules.defaultTeam, pos);
+    }   
```

Comment:
```suggestion
    public Unit spawn(Team team, Position pos){
        return spawn(team, pos.getX(), pos.getY());
    }

    public Unit spawn(Position pos){
        return spawn(state.rules.defaultTeam, pos);
    }   
```
Fix and add compatability for inputting `Vec2`s or anything else that `implements` Position.

Reply:
I'll reset back a commit, recommit while using Position, and force push to avoid having two commits